src/modules/commander/state_machine_helper.cpp

```diff
@@ -582,6 +583,19 @@ bool set_nav_state(vehicle_status_s *status, actuator_armed_s *armed, commander_
 
 		break;
 
+	case commander_state_s::MAIN_STATE_ORBIT:
+		/* require local position */
+		if (status->engine_failure) {
+			status->nav_state = vehicle_status_s::NAVIGATION_STATE_AUTO_LANDENGFAIL;
+
+		} else if (is_armed && check_invalid_pos_nav_state(status, old_failsafe, mavlink_log_pub, status_flags, false, false)) {
+			// nothing to do - everything done in check_invalid_pos_nav_state
+
```

Comment:
What about loss of manual control or data link?

Reply:
Yes, data link! I should "inherit" from loiter, not takeoff.
Manual control is optional for orbit, it can be used to adjust the circle but is not mandatory.